includes/widgets/class-wc-widget-layered-nav.php

```diff
@@ -355,34 +355,83 @@ protected function get_filtered_term_product_counts( $term_ids, $taxonomy, $quer
 			}
 		}
 
-		$meta_query     = new WP_Meta_Query( $meta_query );
-		$tax_query      = new WP_Tax_Query( $tax_query );
-		$meta_query_sql = $meta_query->get_sql( 'post', $wpdb->posts, 'ID' );
-		$tax_query_sql  = $tax_query->get_sql( $wpdb->posts, 'ID' );
-		$term_ids_sql   = '(' . implode( ',', array_map( 'absint', $term_ids ) ) . ')';
+		$hide_out_of_stock = 'yes' === get_option( 'woocommerce_hide_out_of_stock_items' );
```

Comment:
I can appreciate changing this logic but it seems like this logic needs to be abstracted into another function that can be instead be called here. Replacing the custom SQL select query here is complicating this method further :(